aws/src/main/java/org/apache/iceberg/aws/glue/LockManagers.java

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.aws.glue;
+
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.common.DynConstructors;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.util.Tasks;
+
+class LockManagers {
+
+  private static final LockManager LOCK_MANAGER_DEFAULT = new InMemoryLockManager(Maps.newHashMap());
+
+  private LockManagers() {
+  }
+
+  public static LockManager defaultLockManager() {
+    return LOCK_MANAGER_DEFAULT;
+  }
+
+  public static LockManager from(Map<String, String> properties) {
+    if (properties.containsKey(CatalogProperties.LOCK_IMPL)) {
+      return loadLockManager(properties.get(CatalogProperties.LOCK_IMPL), properties);
+    } else {
+      return defaultLockManager();
+    }
+  }
+
+  private static LockManager loadLockManager(String impl, Map<String, String> properties) {
+    DynConstructors.Ctor<LockManager> ctor;
+    try {
+      ctor = DynConstructors.builder(LockManager.class).hiddenImpl(impl).buildChecked();
+    } catch (NoSuchMethodException e) {
+      throw new IllegalArgumentException(String.format(
+          "Cannot initialize LockManager, missing no-arg constructor: %s", impl), e);
+    }
+
+    LockManager lockManager;
+    try {
+      lockManager = ctor.newInstance();
+    } catch (ClassCastException e) {
+      throw new IllegalArgumentException(
+          String.format("Cannot initialize LockManager, %s does not implement LockManager.", impl), e);
+    }
+
+    lockManager.initialize(properties);
+    return lockManager;
+  }
+
+  abstract static class LockManagerBase implements LockManager {
```

Comment:
Nit: in similar cases we use `Base` first, like `BaseLockManager`.

Reply:
Fixed